src/test/java/com/amazonaws/encryptionsdk/model/CiphertextHeadersTest.java

```diff
@@ -314,6 +314,24 @@ private void readUptoNonceLen(final ByteBuffer headerBuff) {
         headerBuff.get();
     }
 
+    @Test(expected = BadCiphertextException.class)
+    public void incompleteCiphertext() {
```

Comment:
This test is currently testing already existing behavior in the CiphertextHeader class (note that if you remove the change you made to ParsedCiphertext, this test will still pass because this test isn't handling any ParsedCiphertext). To test the change you made to ParsedCiphertext, it looks like you'll need to create a new file specifically to test ParsedCiphertext. This will also give you the opportunity to test out the happy case for ParsedCiphertext construction, which is a test that is currently missing.